api/src/main/java/io/minio/MinioClient.java

```diff
@@ -2132,14 +2135,47 @@ public String presignedGetObject(String bucketName, String objectName, Integer e
       throw new InvalidExpiresRangeException(expires, "expires must be in range of 1 to " + DEFAULT_EXPIRY_TIME);
     }
 
+    byte[] body = null;
+    if (method == Method.PUT || method == Method.POST) {
+      body = new byte[0];
+    }
+
     String region = getRegion(bucketName);
-    Request request = createRequest(Method.GET, bucketName, objectName, region,
-                                    null, reqParams, null, null, 0);
+    Request request = createRequest(method, bucketName, objectName, region, null, reqParams, null, body, 0);
     HttpUrl url = Signer.presignV4(request, region, accessKey, secretKey, expires);
     return url.toString();
   }
 
   /**
+   * Returns an presigned URL to download the object in the bucket with given expiry time with custom request params.
+   *
+   * </p><b>Example:</b><br>
+   * <pre>{@code String url = minioClient.presignedGetObject("my-bucketname", "my-objectname", 60 * 60 * 24, reqParams);
+   * System.out.println(url); }</pre>
+   *
+   * @param bucketName  Bucket name.
+   * @param objectName  Object name in the bucket.
+   * @param expires     Expiration time in seconds of presigned URL.
+   * @param reqParams   Override values for set of response headers. Currently supported request parameters are
+   *                    [response-expires, response-content-type, response-cache-control, response-content-disposition]
+   *
+   * @return string contains URL to download the object.
+   *
+   * @throws InvalidBucketNameException   upon an invalid bucket name
+   * @throws InvalidKeyException          upon an invalid access key or secret key
+   * @throws IOException                  upon signature calculation failure
+   * @throws NoSuchAlgorithmException     upon requested algorithm was not found during signature calculation
+   * @throws InvalidExpiresRangeException upon input expires is out of range
+   */
+  public String presignedGetObject(String bucketName, String objectName, Integer expires,
```

Comment:
We can have `getPresignedObjectUrl` method called within `presignedPutObject` exactly the same way  `presignedGetObject` is modified.